pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -639,9 +640,15 @@ func (w *watcher) commitFiles(ctx context.Context, latestData, eventName, commit
 		EventName: eventName,
 	}
 	commitMsg = parseCommitMsg(commitMsg, args)
-	if err := repo.CommitChanges(ctx, repo.GetClonedBranch(), commitMsg, false, changes); err != nil {
+	branch := getBranchName(newBranch, eventName, repo.GetClonedBranch())
+	if err := repo.CommitChanges(ctx, branch, commitMsg, newBranch, changes); err != nil {
 		return fmt.Errorf("failed to perform git commit: %w", err)
 	}
+	if newBranch {
+		if err := repo.Push(ctx, branch); err != nil {
+			return fmt.Errorf("failed to perform git push: %w", err)
+		}
+	}
```

Comment:
I think we should not push to the branch here. This function is named `commitFiles`, which means it only does add changed files to the local branch as the commit step (same as performing git commit command)

In fact, currently we perform pushing to remote repo at the caller side of this function
ref: 
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/eventwatcher/eventwatcher.go#L563
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/eventwatcher/eventwatcher.go#L432

(L432 is handler for in app-config eventwatcher while L563 is handler for the old EventWatcher kind manifest - this one is deprecated).

So we have to reimplement the caller side to make this work smoothly. The current implementation of this PR is a bit tricky.

Reply:
thank you!! I reimplemented the caller side.
[retry push commits in execute func](https://github.com/pipe-cd/pipecd/pull/4395/commits/21cb14b8fd34a9fc537c9ec8fafd45d47ed93a52)